src/federation_server.js

```diff
@@ -0,0 +1,142 @@
+import axios from 'axios';
+import URI from 'URIjs';
+import Promise from 'bluebird';
+import toml from 'toml';
+
+export class FederationServer {
+  /**
+   * FederationServer handles a network connection to a
+   * [federation server](https://www.stellar.org/developers/learn/concepts/federation.html)
+   * instance and exposes an interface for requests to that instance.
+   * @constructor
+   * @param {object} [config] The server configuration.
+   * @param {boolean} [config.secure] Use https, defaults false.
+   * @param {string} [config.hostname] The hostname of the [federation server](https://www.stellar.org/developers/learn/concepts/federation.html), defaults to "localhost".
+   * @param {number} [config.port] Server port, defaults to 80.
+   * @param {string} [config.path] The path to federation endpoint, defaults to "/federation".
+   * @param {string} [config.domain] Domain this federation server represents. Optional, if exist username passed to {@link FederationServer#federation} method does not need to contain a domain.
+   */
+  constructor(config={}) {
+    this.protocol = config.secure ? "https" : "http";
+    this.hostname = config.hostname || "localhost";
+    this.port = config.port || 80;
+    this.path = config.path || '/federation';
+    this.domain = config.domain;
+    this.serverURL = URI({
+      protocol: this.protocol,
+      hostname: this.hostname,
+      port: this.port,
+      path: this.path
+    });
+  }
+
+  /**
+   * This methods splits Stellar address (ex. `bob*stellar.org`) and then tries to find information about federation
+   * server in `stellar.toml` file for a given domain. It returns a `Promise` which resolves if federation server exists
+   * and user has been found and rejects in all other cases.
+   * ```js
+   * StellarSdk.FederationServer.forAddress('bob*stellar.org')
+   *  .then(federationRecord => {
+   *    // {
+   *    //   stellar_address: 'bob*stellar.org',
+   *    //   account_id: 'GB5XVAABEQMY63WTHDQ5RXADGYF345VWMNPTN2GFUDZT57D57ZQTJ7PS'
+   *    // }
+   *  });
+   * ```
+   * This function is here for convenience and is using `createForDomain` and non-static `forAddress` internally.
+   * @see <a href="https://www.stellar.org/developers/learn/concepts/federation.html" target="_blank">Federation doc</a>
+   * @see <a href="https://www.stellar.org/developers/learn/concepts/stellar-toml.html" target="_blank">Stellar.toml doc</a>
+   * @param {string} address Stellar Address (ex. `bob*stellar.org`)
+   * @returns {Promise}
+   */
+  static forAddress(address) {
+    let [,domain] = address.split('*');
```

Comment:
This will behave incorrectly in the presence of 2 asterisks.  We should ensure there isn't, returning an error to the user if there is.